### PR TITLE
Make inclusion of jsoncpp source optional

### PR DESCRIPTION
**Pull request by tonerdo**
Fixes dotnet/aspnetcore#30888

### CODE REVIEW

File: src/signalrclient/CMakeLists.txt [hunk 1]
```diff
@@ -61,6 +60,14 @@ else()
     target_compile_options(signalrclient PRIVATE -Wextra -Wpedantic -Wno-unknown-pragmas)
 endif()
 
+if(INCLUDE_JSONCPP)
+  target_sources(signalrclient PRIVATE ../../third_party_code/jsoncpp/jsoncpp.cpp)
+  target_include_directories(signalrclient PRIVATE ../../third_party_code/jsoncpp)
+else()
+  target_link_libraries(signalrclient PUBLIC ${JSONCPP_LIB})
+  unset(JSONCPP_LIB CACHE)
```
Comment (BrennanConroy) [inline src/signalrclient/CMakeLists.txt:68]: What is this for?

**Review by BrennanConroy — COMMENTED** [review]
Can you add the new variables to the README?
https://github.com/aspnet/SignalR-Client-Cpp/blob/main/README.md#command-line-build

File: src/signalrclient/CMakeLists.txt (hunk 1 above)
Comment (tonerdo) [reply src/signalrclient/CMakeLists.txt]: Hold-over from some testing I was doing. Thanks for catching

**Review by BrennanConroy — APPROVED** [review]

**Comment by BrennanConroy** [comment]
Awesome, thanks for the change!

**Pull request merged**